mingw-w64-x86_64-qgpgme-qt6: CMake module is non relocatable

### Description / Steps to reproduce the issue

CMakeLists.txt:

```cmake
cmake_minimum_required(VERSION 3.19)
project(gpgme2 C CXX)

find_package(Qt6 COMPONENTS Core REQUIRED)
find_package(QGpgmeQt6 REQUIRED)

add_executable(main main.cpp)

target_link_libraries(main PUBLIC QGpgmeQt6 )
```

main.cpp:

```cpp
int main() {
    return 0;
}
```

1. With the above files proceed to configure.
2. It will error out because `QGpgQt6Config.cmake` is not relocatable at all, unlike its qt5 counterpart:

```

# Create imported target QGpgmeQt6
add_library(QGpgmeQt6 SHARED IMPORTED)

set_target_properties(QGpgmeQt6 PROPERTIES
  IMPORTED_IMPLIB_RELEASE "/clang64/lib/libqgpgmeqt6.dll.a"
  INTERFACE_INCLUDE_DIRECTORIES "/clang64/include/qgpgme;/clang64/include"
  INTERFACE_LINK_LIBRARIES "Gpgmepp;Qt6::Core"
  IMPORTED_LOCATION "/clang64/bin/libqgpgmeqt6-15.dll"
)
get_target_property(_libpath QGpgmeQt6 IMPORTED_IMPLIB_RELEASE)
get_target_property(_dllpath QGpgmeQt6 IMPORTED_LOCATION)

```

### Expected behavior

Configuration should succeed.

### Actual behavior

CMake errors out:

```
[main] Configurando el proyecto: gpgme 
[driver] Quitando c:/Users/Amalia/Desktop/Rachel/gpgme/build/CMakeCache.txt
[driver] Quitando c:\Users\Amalia\Desktop\Rachel\gpgme\build\CMakeFiles
[proc] Ejecutando el comando: D:/msys64/clang64/bin/cmake.exe -DCMAKE_BUILD_TYPE:STRING=Debug -DCMAKE_INSTALL_PREFIX:STRING=c:/Users/Amalia/Desktop/Rachel/gpgme/install -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE -DCMAKE_CXX_COMPILER:FILEPATH=D:/msys64/clang64/bin/clang++.exe -DCMAKE_CC_COMPILER:FILEPATH=D:/msys64/clang64/bin/clang.exe --no-warn-unused-cli -SC:/Users/Amalia/Desktop/Rachel/gpgme -Bc:/Users/Amalia/Desktop/Rachel/gpgme/build -G Ninja
[cmake] Not searching for unused variables given on the command line.
[cmake] -- The C compiler identification is Clang 18.1.8
[cmake] -- The CXX compiler identification is Clang 18.1.8
[cmake] -- Detecting C compiler ABI info
[cmake] -- Detecting C compiler ABI info - done
[cmake] -- Check for working C compiler: D:/msys64/clang64/bin/cc.exe - skipped
[cmake] -- Detecting C compile features
[cmake] -- Detecting C compile features - done
[cmake] -- Detecting CXX compiler ABI info
[cmake] -- Detecting CXX compiler ABI info - done
[cmake] -- Check for working CXX compiler: D:/msys64/clang64/bin/clang++.exe - skipped
[cmake] -- Detecting CXX compile features
[cmake] -- Detecting CXX compile features - done
[cmake] -- Performing Test CMAKE_HAVE_LIBC_PTHREAD
[cmake] -- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
[cmake] -- Check if compiler accepts -pthread
[cmake] -- Check if compiler accepts -pthread - yes
[cmake] -- Found Threads: TRUE
[cmake] -- Performing Test HAVE_STDATOMIC
[cmake] -- Performing Test HAVE_STDATOMIC - Success
[cmake] -- Found WrapAtomic: TRUE
[cmake] CMake Error at D:/msys64/clang64/lib/cmake/QGpgmeQt6/QGpgmeQt6Config.cmake:84 (message):
[cmake]   The imported target "QGpgmeQt6" references the file
[cmake] 
[cmake]      "/clang64/lib/libqgpgmeqt6.dll.a"
[cmake] 
[cmake]   but this file does not exist.  Possible reasons include:
[cmake] 
[cmake]   * The file was deleted, renamed, or moved to another location.
[cmake] 
[cmake]   * An install or uninstall procedure did not complete successfully.
[cmake] 
[cmake]   * The installation package was faulty and contained
[cmake] 
[cmake]      "D:/msys64/clang64/lib/cmake/QGpgmeQt6/QGpgmeQt6Config.cmake"
[cmake] 
[cmake]   but not all the files it references.
[cmake] 
[cmake] Call Stack (most recent call first):
[cmake]   CMakeLists.txt:5 (find_package)
[cmake] 
[cmake] 
[cmake] -- Configuring incomplete, errors occurred!
[proc] El comando: D:/msys64/clang64/bin/cmake.exe -DCMAKE_BUILD_TYPE:STRING=Debug -DCMAKE_INSTALL_PREFIX:STRING=c:/Users/Amalia/Desktop/Rachel/gpgme/install -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE -DCMAKE_CXX_COMPILER:FILEPATH=D:/msys64/clang64/bin/clang++.exe -DCMAKE_CC_COMPILER:FILEPATH=D:/msys64/clang64/bin/clang.exe --no-warn-unused-cli -SC:/Users/Amalia/Desktop/Rachel/gpgme -Bc:/Users/Amalia/Desktop/Rachel/gpgme/build -G Ninja salió con el código: 1
```

### Verification

- [X] I have verified that my MSYS2 is up-to-date before submitting the report (see https://www.msys2.org/docs/updating/)

### Windows Version

MINGW64_NT-10.0-19045

### MINGW environments affected

- [ ] MINGW64
- [ ] MINGW32
- [ ] UCRT64
- [X] CLANG64
- [ ] CLANGARM64

### Are you willing to submit a PR?

_No response_

--- a/mingw-w64-gpgme/PKGBUILD
+++ b/mingw-w64-gpgme/PKGBUILD
@@ -10,7 +10,7 @@ pkgname=("${MINGW_PACKAGE_PREFIX}-${_realname}"
            "${MINGW_PACKAGE_PREFIX}-qgpgme-qt6" \
            "${MINGW_PACKAGE_PREFIX}-python-gpgme"))
 pkgver=1.23.2
-pkgrel=8
+pkgrel=9
 pkgdesc="A C wrapper library for GnuPG (mingw-w64)"
 arch=('any')
 mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64' 'clangarm64')
@@ -66,7 +66,7 @@ sha256sums=('9499e8b1f33cccb6815527a1bc16049d35a6198a6c5fae0185f2bd561bce5224'
             '0b0d511f9d47844d3fd806864310d9abbeb1285d43f6ffa33babf55a4a410458'
             'ace3d5166372e0422a625f76a2890d70f2916d9239ade74e58bd23077e9c1c6b'
             '5e8e7e773a09ed1e1724c9d7216eaaecce52af49152513366026a2fa3245d762'
-            '81f412ffdeeb2a0757248b517fbda9caae9913b1552c88abb8347faa4d1efea2'
+            '2a390489e8d192ca1b182458f29cbc5d0b78524b041afeee91bde6001150b5c2'
             'a110b40d146a7c561140b68831027f61d2132e1069d2476b2e0a4034bc54fae1')
 
 _apply_patch_with_msg() {

--- a/mingw-w64-gpgme/relocatable-cmake.patch
+++ b/mingw-w64-gpgme/relocatable-cmake.patch
@@ -1,5 +1,4 @@
-diff --git a/lang/cpp/src/GpgmeppConfig-w32.cmake.in.in b/lang/cpp/src/GpgmeppCo
-nfig-w32.cmake.in.in
+diff --git a/lang/cpp/src/GpgmeppConfig-w32.cmake.in.in b/lang/cpp/src/GpgmeppConfig-w32.cmake.in.in
 index 12826760..df1d4c9a 100644
 --- a/lang/cpp/src/GpgmeppConfig-w32.cmake.in.in
 +++ b/lang/cpp/src/GpgmeppConfig-w32.cmake.in.in
@@ -69,6 +68,45 @@ index b8978053..8f7a2d43 100644
 @@ -73,6 +81,8 @@
  list(APPEND _IMPORT_CHECK_TARGETS QGpgme )
  list(APPEND _IMPORT_CHECK_FILES_FOR_QGpgme "${_libpath}" "${_dllpath}" )
+ 
++set(_IMPORT_PREFIX)
++
+ if(CMAKE_VERSION VERSION_LESS 2.8.12)
+   message(FATAL_ERROR "This file relies on consumers using CMake 2.8.12 or greater.")
+ endif()
+diff --git a/lang/qt/src/QGpgmeQt6Config-w32.cmake.in.in b/lang/qt/src/QGpgmeQt6Config-w32.cmake.in.in
+index 290f211..bb2df7a 100644
+--- a/lang/qt/src/QGpgmeQt6Config-w32.cmake.in.in
++++ b/lang/qt/src/QGpgmeQt6Config-w32.cmake.in.in
+@@ -58,14 +58,22 @@
+ unset(_targetsNotDefined)
+ unset(_expectedTargets)
+ 
++get_filename_component(_IMPORT_PREFIX "${CMAKE_CURRENT_LIST_FILE}" PATH)
++get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
++get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
++get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
++if(_IMPORT_PREFIX STREQUAL "/")
++  set(_IMPORT_PREFIX "")
++endif()
++
+ # Create imported target QGpgmeQt6
+ add_library(QGpgmeQt6 SHARED IMPORTED)
+ 
+ set_target_properties(QGpgmeQt6 PROPERTIES
+-  IMPORTED_IMPLIB_RELEASE "@resolved_libdir@/libqgpgmeqt6.dll.a"
+-  INTERFACE_INCLUDE_DIRECTORIES "@resolved_includedir@/qgpgme;@resolved_includedir@"
++  IMPORTED_IMPLIB_RELEASE "${_IMPORT_PREFIX}/lib/libqgpgmeqt6.dll.a"
++  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include/qgpgme;${_IMPORT_PREFIX}/include"
+   INTERFACE_LINK_LIBRARIES "Gpgmepp;Qt6::Core"
+-  IMPORTED_LOCATION "@resolved_bindir@/libqgpgmeqt6-15.dll"
++  IMPORTED_LOCATION "${_IMPORT_PREFIX}/bin/libqgpgmeqt6-15.dll"
+ )
+ get_target_property(_libpath QGpgmeQt6 IMPORTED_IMPLIB_RELEASE)
+ get_target_property(_dllpath QGpgmeQt6 IMPORTED_LOCATION)
+@@ -73,6 +81,8 @@
+ list(APPEND _IMPORT_CHECK_TARGETS QGpgmeQt6 )
+ list(APPEND _IMPORT_CHECK_FILES_FOR_QGpgmeQt6 "${_libpath}" "${_dllpath}" )
  
 +set(_IMPORT_PREFIX)
 +

